romeo/src/state.rs

```diff
@@ -232,10 +238,23 @@ fn process_bitcoin_block(
 	bitcoin_height: u32,
 	block: Block,
 ) -> (State, Vec<Task>) {
-	state
-		.deposits
-		.extend(parse_deposits(config, bitcoin_height, &block));
-	state.withdrawals.extend(parse_withdrawals(config, &block));
+	state.deposits.extend(
+		parse_deposits(config, bitcoin_height, &block)
+			.into_iter()
+			.filter(|deposit| {
+				deposit.info.sbtc_wallet_address == config.sbtc_wallet_address()
```

Comment:
Consider storing `config.sbtc_wallet_address()` before iterating.

Reply:
Addressed in 41854d9f6ae44a968b7fdb24184dd85041492503